awswrangler/distributed/ray/modin/_core.py

```diff
@@ -1,12 +1,48 @@
 """Modin on Ray Core module (PRIVATE)."""
 # pylint: disable=import-outside-toplevel
+import logging
 from functools import wraps
-from typing import Any, Callable, Optional
+from typing import Any, Callable, List, Optional
 
 import pandas as pd
+import ray
 from modin.distributed.dataframe.pandas import from_partitions, unwrap_partitions
 from modin.pandas import DataFrame as ModinDataFrame
 
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+def _validate_partition_cols(df: pd.DataFrame) -> bool:
+    """
+    Validate if partitions of the data frame are partitioned along row axis.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Modin data frame
+
+    Returns
+    -------
+    bool
+    """
+
+    @ray.remote
+    def _validate_partition(df: pd.DataFrame, n_columns: int) -> bool:
+        return len(df.columns) == n_columns
+
+    # Unwrap partitions as they are currently stored (axis=None)
+    # Partitions are a 2D array because the data frame is split along both row and column axis
+    partitions: List[List[ray.types.ObjectRef[pd.DataFrame]]] = unwrap_partitions(df, axis=None)
+    return all(
```

Comment:
I might be wrong, but wouldn't just be the equivalent of checking that the partition shape is `1` along the column axis like [here](https://modin.readthedocs.io/en/stable/usage_guide/optimization_notes/index.html#how-to-tune-partitioning)
```
partitions_shape = np.array(unwrap_partitions(df)).shape
return partitions_shape[1] == 1
```

Reply:
Kind of but not exactly. The shape might be right, but what if the partition doesn't contain all columns?

I do very much like how concise it is though so if you think the concern above isn't valid happy to put that in instead